dynamic/run.sh

```diff
@@ -94,51 +94,132 @@ rm -rf /tmp/* /opt/instana/agent/etc/org.ops4j.pax.logging.cfg \
+gomplate < /root/org.ops4j.pax.logging.cfg.tmpl > /opt/instana/agent/etc/org.ops4j.pax.logging.cfg
+gomplate < /root/org.ops4j.pax.url.mvn.cfg.tmpl > /opt/instana/agent/etc/org.ops4j.pax.url.mvn.cfg
+gomplate < /root/mvn-settings.xml.tmpl > /opt/instana/agent/etc/mvn-settings.xml
+gomplate < /root/com.instana.agent.main.sender.Backend-1.cfg.tmpl > \
   /opt/instana/agent/etc/instana/com.instana.agent.main.sender.Backend-1.cfg
-cat /root/com.instana.agent.bootstrap.AgentBootstrap.cfg.tmpl | gomplate > \
+gomplate < /root/com.instana.agent.bootstrap.AgentBootstrap.cfg.tmpl > \
   /opt/instana/agent/etc/instana/com.instana.agent.bootstrap.AgentBootstrap.cfg
-cat /root/com.instana.agent.main.config.UpdateManager.cfg.tmpl | gomplate > \
+gomplate < /root/com.instana.agent.main.config.UpdateManager.cfg.tmpl > \
   /opt/instana/agent/etc/instana/com.instana.agent.main.config.UpdateManager.cfg
 
-if [ ! -z "${INSTANA_AGENT_HTTP_LISTEN}" ]; then
+if [ -n "${INSTANA_AGENT_HTTP_LISTEN}" ]; then
   echo -e "\nhttp.listen = ${INSTANA_AGENT_HTTP_LISTEN}" >> /opt/instana/agent/etc/instana/com.instana.agent.main.config.Agent.cfg
 fi
 
-if [ ! -z "${INSTANA_AGENT_MODE}" ]; then
-  if [ "${INSTANA_AGENT_MODE}" = "AWS" ]; then
+if [ "${INSTANA_AGENT_MODE}" = 'AWS' ]; then
+  echo 'AWS mode configured'
+
+  ###
+  # Discover which platform we are running on
+  ###
+  if [ -n "${ECS_CONTAINER_METADATA_URI}" ]; then
+    PLATFORM='ECS'
+    echo 'Running on ECS'
+  elif curl -s http://169.254.169.254/latest/meta-data/ --connect-timeout 2 --fail 2>&1 /dev/null; then
+    PLATFORM='EC2'
+    echo 'Running on EC2'
+  else
+    PLATFORM='UNKNOWN'
+    echo 'Cannot recognize the platform; it is neither ECS nor EC2'
+  fi
 
-    INSTANA_AWS_REGION_CONFIG=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document --connect-timeout 2 | awk -F\" '/region/ {print $4}')
+  ###
+  # Retrieve region
+  ###
+  if [ -n "${INSTANA_AWS_REGION_CONFIG}" ]; then
+    echo "AWS region configured via environment: ${INSTANA_AWS_REGION_CONFIG}"
+  else
+    case "${PLATFORM}" in
 
-    if [ $? != 0 ]; then
-      log_error "Error querying AWS metadata."
-      exit 1
-    fi
+    'ECS')
+      if ! curl -s "${ECS_CONTAINER_METADATA_URI}/task" --connect-timeout 2 --fail -o /tmp/ecs_data; then
+        echo "Cannot retrieve metadata from the '${ECS_CONTAINER_METADATA_URI}/task' endpoint. Aborting startup."
+        exit 1
+      fi
 
-    export INSTANA_AWS_REGION_CONFIG
+      if AWS_AVAILABILITY_ZONE=$(jq -e -r '.AvailabilityZone' < /tmp/ecs_data); then
+        # The -e flag of jq will make it fail on purpose, if the data do not contain the 'AvailabilityZone' key
+        readonly ARN_REGEXP='^([a-z]+-[a-z]+-[0-9])*'
+        if [[ "${AWS_AVAILABILITY_ZONE}" =~ ${ARN_REGEXP} ]]; then
+          echo "AWS region parsed from the availability zone"
+          INSTANA_AWS_REGION_CONFIG="${BASH_REMATCH[1]}"
```

Comment:
Change these lines around and directly log the region config we have found? `"...availability zone: ${INSTANA_AWS_REGION_CONFIG}"`